test/common.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-console */
+import React from 'react'
+import {createStore} from 'redux'
+import {Provider} from 'react-redux'
+import PropTypes from 'prop-types'
+import {Promise} from 'bluebird'
+
+export const GET_DATA_DELAY = 0.1 * 1000
+
+export const getData = (data, msDelay = GET_DATA_DELAY) => (
+  Promise.delay(msDelay).then(() => data)
+)
+
+export function getDataWithCount({data}, msDelay) {
+  return getData({data: data + ++getDataWithCount.counter}, msDelay)
```

Comment:
Hmm, do we need this meta-information here? Can't we just check the counter?

Reply:
I thought about it, but then I decided that, since these are "integration" tests, we really want to test the functionality as a whole, so we need to check if the component renders the data correctly. In order to test that, I need to update the data accordingly, i.e. change the data on each `getData` call...